components/TextFields/tests/unit/ContainedInputView/MDCBaseTextFieldTests.m

```diff
@@ -184,4 +184,67 @@ - (void)testClearButtonRectForBounds {
   XCTAssertTrue(CGRectEqualToRect(actualClearButtonFrame, expectedClearButtonFrame));
 }
 
+- (void)testFloatingLabelColorAccessors {
+  // Given
+  CGRect textFieldFrame = CGRectMake(0, 0, 130, 40);
+  MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:textFieldFrame];
+  UIColor *floatingLabelColor = [UIColor redColor];
+
+  // When
+  [textField setFloatingLabelColor:floatingLabelColor forState:UIControlStateNormal];
+  [textField setFloatingLabelColor:floatingLabelColor forState:MDCTextControlStateEditing];
+  [textField setFloatingLabelColor:floatingLabelColor forState:UIControlStateDisabled];
+  [textField setNeedsLayout];
+  [textField layoutIfNeeded];
+
+  // Then
+  XCTAssertEqualObjects(floatingLabelColor,
+                        [textField floatingLabelColorForState:UIControlStateNormal]);
+  XCTAssertEqualObjects(floatingLabelColor,
+                        [textField floatingLabelColorForState:MDCTextControlStateEditing]);
+  XCTAssertEqualObjects(floatingLabelColor,
+                        [textField floatingLabelColorForState:UIControlStateDisabled]);
+}
```

Comment:
For the most complicated use case we want to make sure that we do the same thing that UIButton does for all the states we support.

Here I would create a local UIButton and expect that the same combinations work equal for our custom stateful object. (we only care about the 3 explained in the header so an array would be useful here too)

Reference test in MDCButton:
https://github.com/material-components/material-components-ios/blob/e3784afe259055679479f09b2e92324b61cf7fe0/components/Buttons/tests/unit/MDCButtonTests.m#L461

Reply:
Hmmm, so what kind of fallback behavior do we want to support exactly?

In general, I don't think we should try to treat these TextFields like they're UIButtons. The weird thing about UIControlState is that every UIControl treats it differently. UITextFields treat UIControlState totally differently than UIButtons do. Where a UIButton might actually look and behave differently when it's highlighted or selected, a UITextField won't. This screenshot of a bunch of UITextFields in different states shows that only disabled makes a visual difference:
<img width="140" alt="Screen Shot 2019-09-05 at 10 00 49 AM" src="https://user-images.githubusercontent.com/8020010/64567611-9f84a880-d326-11e9-9af4-0c171a7496b7.png">
The way it is right now there are default values for the three supported UIControlStates (i.e. the three that are adapted to underlying MDCContainedInputViewStates), and the stateful accessors for those states only accept nonnull colors, so clients will not be able to set nil for a given state. All this makes me wonder how much we really need to think about fallback behavior.